daemon/create.go

```diff
@@ -14,6 +14,16 @@ import (
 	"github.com/opencontainers/runc/libcontainer/label"
 )
 
+// ContainerCreateAPI takes configs and creates a container, returning
+// the ID of the newly created container.
+func (daemon *Daemon) ContainerCreateAPI(name string, config *runconfig.Config, hostConfig *runconfig.HostConfig, adjustCPUShares bool) (string, []string, error) {
```

Comment:
Yea, let's just make it return the ID and the builder can look it up. The small performance hit isn't an issue in the builder's case.  Then we can get ride of this (near) dup func.